40.md

```diff
@@ -0,0 +1,46 @@
+NIP-40
+======
+
+Expiration Timestamp
+-----------------------------------
+
+`draft` `optional` `author:0xtlt`
+
+The `expiration` tag enables users to specify a timestamp at which the message should be considered expired and deleted by relays. This helps ensure that messages are not retained by the network after they are no longer relevant, which helps maintain the privacy of the messages. This is especially useful in cases where a replacement event is not available or feasible.
+
+#### Spec
+
+```
+tag: expiration
+options:
+ - [timestamp]: required
+```
+
+#### Example
+
+```json
+{
+    "pubkey": "<pub-key>",
+    "created_at": 1000000000,
+    "kind": 1,
+    "tags": [
+      ["expiration", 1600000000]
+    ],
+    "content": "This message will expire at the specified timestamp and be deleted by relays.\n",
+    "id": "<event-id>"
+}
+```
+
+Note: The timestamp should be in the same format as the created_at timestamp and should be interpreted as the time at which the message should be deleted by relays.
+
+Client Behavior
+---------------
+
+Clients SHOULD use the `supported_nips` field to learn if a relay supports this NIP. Clients SHOULD NOT send expiration events to relays that do not support this NIP.
```

Comment:
Relays oblivious to this nip will not care about such desires and just mirror these events. A nip can only impose stuff on software supporting the nip, not others. I wouldn't even imply there were any effort to keep these events away from relays that don't support the nip.

Reply:
Was already "implemented" for nip16 : https://github.com/nostr-protocol/nips/blob/master/16.md

But I understand what you have in mind.

It's not a requirement but a recommendation here